.travis.yml

```diff
@@ -43,7 +43,7 @@ jobs:
       env:
       cache: false
       language: python
-      python: '3.6'
+      python: '3.4' # Oldest supported version according to doc/dependencies.md
```

Comment:
The linter uses the available Python executable to lint the functional tests. It will throw an error if sees a syntax that it doesn't recognise, see above demo. So it's not necessary to run the functional tests themselves in Python 3.4, which would require adding pyenv as a dependency inside the Docker container and slow things down.

Reply:
Also, though hopefully not needed, any developer who uses pyenv on their machine will detect any mistake that slips through the cracks once they run the functional test suite on master.